### ISSUE TITLE
lgrep_curbuf command issue

### ISSUE THREAD

**Issue by hqkhan** [ibhagwan/fzf-lua#566]
### Info

- Operating System: Linux
- Shell: bash
- Terminal: iterm2
- `nvim --version`: 0.7.2
- `fzf --version`: 0.30.0 (2093667)

### Description 

Problem #1
Command `lgrep_curbuf` currently does not work if you give it a function for `opts` as you can with `grep_curbuf` for example. Here is relevant config for this problem:

```lua
local map_fzf = function(mode, key, f, options, buffer)

  local desc = nil
  if type(options) == 'table' then
    desc = options.desc
    options.desc = nil
  elseif type(options) == 'function' then
    desc = options().desc
  end

  local rhs = function()
    if not pcall(require, 'fzf-lua') then
      require('packer').loader('fzf-lua')
    end
    require('plugins.fzf-lua')[f](options or {})
  end

  local map_options = {
    silent = true,
    buffer = buffer,
    desc   = desc or string.format("FzfLua %s", f),
  }

  vim.keymap.set(mode, key, rhs, map_options)
end

-- Live grep cur buf with function (ERROR)
map_fzf('n', "<leader>lg", "lgrep_curbuf",
    function() return { desc = "Live grep",
        winopts = {
            preview = { vertical = "down:65%", horizontal = "right:75%", }
        },
        search=vim.fn.expand("<cword>"),
} end)
```

This will produce an error from `grep.lua` as `lgrep_curbuf` function doesn't support `opts` being a function. This can be seen [here](https://github.com/ibhagwan/fzf-lua/blob/664358dc45eb4603284ddbfad308a454f6536638/lua/fzf-lua/providers/grep.lua#L472). 

Problem #2

`search` given in `opts` is not respected if the command is `lgrep_curbuf`. Here's what I've been able to find. This seems to be because `opts.search` is never looked at [here](https://github.com/ibhagwan/fzf-lua/blob/664358dc45eb4603284ddbfad308a454f6536638/lua/fzf-lua/providers/grep.lua#L459)

```
-- Live grep cur buf with dict (Search ignored)
map_fzf('n', "<leader>lG", "lgrep_curbuf",      
{ desc = "Live grep (buffer)", prompt = 'Buffer❯ ', 
search=vim.fn.expand("<cword>")})
```

The above command will simply be livegrep without getting kickstarted by `search` parameter.

I also took a look at the `README` but wasn't able to find anything documenting `search` as a possible option to pass into opts. I think adding a one liner about this option would be great as I find this to be extremely useful.

**Comment by ibhagwan** [comment]
https://github.com/ibhagwan/fzf-lua/commit/2a6af4812948cb69fc66c57a07b01411a0e2ba6f - this commit fixes problem 1.

As for problem 2: I just tried it and it works, this is the output from running `:FzfLua lgrep_curbuf search=debug`:
![image](https://user-images.githubusercontent.com/59988195/201099312-e4ab176b-02da-48a5-bf72-9e7ea50b7612.png)

From there I can `<C-g>` my way into `grep_curbuf` and vice versa.

The only reason `opts.search` is being modified below is because if the user didn't specify a search term for `grep_curbuf`, they will be presented with the `Grep For>` prompt which I felt wasn't warranted for the current buffer (as you probably want to fuzzy all lines), providing an empty search string avoids the input prompt (this is what `grep_project` does):
https://github.com/ibhagwan/fzf-lua/blob/664358dc45eb4603284ddbfad308a454f6536638/lua/fzf-lua/providers/grep.lua#L462-L463

Now the reason why your code doesn't work:
```lua
map_fzf('n', "<leader>lG", "lgrep_curbuf",      
{ desc = "Live grep (buffer)", prompt = 'Buffer❯ ',  search=vim.fn.expand("<cword>")})
```

If we take a look at what `map_fzf` does we can see it saves the `rhs` part as a function in the mapping:
```lua
  local rhs = function()
    if not pcall(require, 'fzf-lua') then
      require('packer').loader('fzf-lua')
    end
    require('plugins.fzf-lua')[f](options or {})
  end
  ...
  vim.keymap.set(mode, key, rhs, map_options)
```

That means that `<cword>` is stored in the mapping when this file is loaded (most likely empty unless you reloaded nvim while the cursor is on a word),  and that is the main reason I added support for options as a function, you want to be retrieving `<cword>` again **with each call to `lgrep_curbuf`**.

TL;DR, update to the latest commit and change your code to the below and everything will work :-)
```lua
map_fzf('n', '<leader>lG', "lgrep_curbuf", function()
  return {
    desc = 'grep <word> (buffer)',
    prompt = 'Buffer❯ ',
    search=vim.fn.expand("<cword>"),
  }
end)
```

> I also took a look at the `README` but wasn't able to find anything documenting `search` as a possible option to pass into opts. I think adding a one liner about this option would be great as I find this to be extremely useful.

You're right, should probably be in the Wiki/README, there are quite a few undocumented options/functionalities in this plugin due to the fact I always prioritized adding new stuff and accomodating user requests over documentation, that said, documentation is in a pretty decent state given a sole maintainer so I'm not too worried about it :-)
